go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/external-secrets/external-secrets
 
-go 1.23
+go 1.23.0
+
+toolchain go1.23.1
```

Comment:
This is still weird. I don't understand why check-diff is adding this. Especially since the version is `1.23.0` but the toolchain is `1.23.1`... makes no sense. The new `vault-cli` needs `1.23.0` so that's the reason for the version change, but not the toolchain addition 🤔

Reply:
Okay, so I ran check-diff and for me it didn't add the toolchain part. Can you try deleting the `toolchain` command but leave the version at 1.23.0 and run make check-diff and see if it added it back? If not, great, commit it. :)

Reply:
I removed the toolchain line and ran the command again and now it does not add it again. I see in my commit [6dea3c374250cb6b25d6ea86dbe009f3af13f62b](https://github.com/external-secrets/external-secrets/pull/3916/commits/6dea3c374250cb6b25d6ea86dbe009f3af13f62b) that it set it to 1.23.0 instead of 1.23, maybe that combination made it go haywire. 
It is gone now at least :)